cabal-install-solver/src/Distribution/Solver/Types/ResolverPackage.hs

```diff
@@ -23,30 +25,36 @@ import Distribution.Simple.Utils (ordNub)
 --
 -- This is like the 'InstallPlan.PlanPackage' but with fewer cases.
 --
-data ResolverPackage loc = PreExisting InstSolverPackage
-                         | Configured  (SolverPackage loc)
+-- The constraint scope field refers to the scope in which this package was resolved.
+data ResolverPackage loc = PreExisting InstSolverPackage ConstraintScope
+                         | Configured  (SolverPackage loc) ConstraintScope
```

Comment:
The packages that are part of the plan now also store information about the scope they were resolved in. This information is propagated down to cabal-install (see `GenericInstallPlan`).